### PR TITLE
Add `platform` option to `sigh`.

### PR DESCRIPTION
**Pull request by fabiomassimo**
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:
- [x] Run `rspec` for all tools you modified
- [x] Run `rubocop -a` to ensure the code style is valid
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:
# Summary

Add a new option to `sigh` to specify the platform of the provisioning profile to manage.
# Why

tvOS is managed in the developer portal as a `sub_platform` that requires its own provisioning profile (before was just iOS).

In this pull request I've added this new option to improve the `sigh`'s support for all available platform and finally take advantage of it for `match` (#5655).
# Notes

Once this has been accepted I'll make a new PR for `match`.

### CODE REVIEW

File: sigh/lib/sigh/options.rb [hunk 1]
```diff
@@ -101,7 +101,17 @@ def self.available_options
                                      env_name: "SIGH_SKIP_CERTIFICATE_VERIFICATION",
                                      description: "Skips the verification of the certificates for every existing profiles. This will make sure the provisioning profile can be used on the local machine",
                                      is_string: false,
-                                     default_value: false)
+                                     default_value: false),
+        FastlaneCore::ConfigItem.new(key: :platform,
+                                     short_option: '-p',
+                                     env_name: "SIGH_PLATFORM",
+                                     description: "Set the provisioning profile's platform (i.e. iOS, tvOS)",
```
Comment (KrauseFx) [inline sigh/lib/sigh/options.rb]: Should be lower case `ios` and `tvos` here, right?

File: sigh/lib/sigh/options.rb [hunk 2]
```diff
@@ -101,7 +101,17 @@ def self.available_options
...
+                                     is_string: true,
+                                     default_value: "ios",
+                                     verify_block: proc do |value|
+                                       pt = %w(macos tvos ios)
```
Comment (KrauseFx) [inline sigh/lib/sigh/options.rb]: I'd love if `value` could also be a symbol, to support it, just add
```ruby
value = value.to_s
```

Comment (KrauseFx) [inline sigh/lib/sigh/options.rb]: This will change the actual value

**Review by KrauseFx — DISMISSED** [review]
Looks good, I added some comments before it's ready to be merged

File: sigh/lib/sigh/options.rb (hunk 2 above)
Comment (fabiomassimo) [inline sigh/lib/sigh/options.rb]: Does it mean that by just adding 

```ruby
value = value.to_s
```

in the `verify_block` it gets automatically passed as string when it is used by `Sigh.config[:platform]` ?

File: sigh/lib/sigh/runner.rb [hunk 3]
```diff
@@ -134,7 +134,9 @@ def create_profile!
       UI.important "Creating new provisioning profile for '#{Sigh.config[:app_identifier]}' with name '#{name}'"
```
Comment (bassrock) [inline sigh/lib/sigh/runner.rb]: Should the platform be added to this description since a tvOS app could share the same app_identifier as an iOS app

**Comment by ohwutup** [comment]
Hey @fabiomassimo! Looks like there are some broken tests on this branch. Would you mind taking a look at that before we can merge?

File: sigh/lib/sigh/runner.rb (hunk 3 above)
Comment (fabiomassimo) [inline sigh/lib/sigh/runner.rb]: Good one. I have added the platform name in the output message.

**Comment by KrauseFx** [comment]
Thanks @fabiomassimo for updating the PR :+1:

**Review by asfalcone — APPROVED** [review]

**Pull request merged**